File: android-reference/app/src/main/java/com/varmeta/promptenhancer/tokenizer/Tokenizer.kt
package com.varmeta.promptenhancer.tokenizer

interface Tokenizer {
    val eosTokenId: Int
    val padTokenId: Int
    val decoderStartTokenId: Int

    fun encode(text: String, maxSourceTokens: Int): IntArray
    fun decode(tokenIds: IntArray): String
}

File: ios-reference/Sources/PromptEnhancer/Tokenizer.swift
import Foundation

public protocol Tokenizer: Sendable {
    var eosTokenId: Int { get }
    var padTokenId: Int { get }
    var decoderStartTokenId: Int { get }

    func encode(_ text: String, maxSourceTokens: Int) -> [Int]
    func decode(_ tokenIds: [Int]) -> String
}
